pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -2033,6 +2063,32 @@ func rktRun(domainName string, ContainerImageID string, xenCfgFilename string, a
 	return domainID, podUUID, nil
 }
 
+// rktStatus checks the status of the pod
+func rktStatus(podUUID string) string {
+	cmd := "rkt"
+	args := []string{
+		"--dir=" + types.PersistRktDataDir,
+		"status",
+		podUUID,
+		"--format=json",
+	}
+	log.Infof("Calling command %s %v\n", cmd, args)
+	stdoutStderr, err := wrap.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		log.Errorln("rkt status failed ", err)
+		log.Errorln("rkt status output ", string(stdoutStderr))
+		return ""
+	}
+	log.Infof("rkt status done\n")
+	var status map[string]interface{}
+	if err := json.Unmarshal(stdoutStderr, &status); err != nil {
+		log.Errorln("rkt status failed ", err)
+		log.Errorln("rkt status output ", string(stdoutStderr))
+		return ""
+	}
+	return status["state"].(string)
```

Comment:
What is the logic here? is the stdoutStderr in json format??

Reply:
While doing rkt status we are passing --formay=json and it will return the state of the pod. If the state is not running then we are returning an error otherwise not.

Reply:
Here, you are assuming "state" key will always be present.. You need to make that check first..

return status["state"].(string)

state, ok := status["state"]
if ok  {
  return state.(string)
}
// No "state" in output.. return error

Reply:
In rkt status, the state key will always be present. So, I think we don't need this check.